PyRDF/backend/Dist.py

```diff
@@ -6,41 +6,21 @@
 import warnings
 import ROOT
 import numpy
+import logging
+from collections import namedtuple
 
+logger = logging.getLogger(__name__)
 
-class Range(object):
-    """
-    Base class to represent ranges.
+Range = namedtuple("Range", ["start", "end", "filelist", "friend_info"])
 
-    A range represents a logical partition of the entries of a chain and is
-    the basis for parallelization. First entry of the range (start) is
-    inclusive while the second one is not (end).
-    """
 
-    def __init__(self, start, end, filelist=None, friend_info=None):
-        """
-        Create an instance of a Range
-
-        Args:
-            start (int): First entry of the range.
-
-            end (int): Last entry of the range, which is exclusive.
-
-            filelist (list, optional): Files where the range of entries
-                belongs to.
-        """
-        self.start = start
-        self.end = end
-        self.filelist = filelist
-        self.friend_info = friend_info
-
-    def __repr__(self):
-        """Return a string representation of the range composition."""
-        if self.filelist:
-            return ("(" + str(self.start) + "," + str(self.end) + "), " +
-                    str(self.filelist))
-        else:
-            return "(" + str(self.start) + "," + str(self.end) + ")"
+def _n_even_chunks(iterable, n_chunks):
+    """Yield `n_chunks` as even chunks as possible from `iterable`."""
+    last = 0
+    for i in range(1, n_chunks + 1):
+        cur = int(round(i * (len(iterable) / n_chunks)))
```

Comment:
Minor detail: Since `len(iterable)` is constant, you can pull it out of the loop to prevent that it's recomputed over and over again.

Reply:
Why this has to be a generator? You could also just return the list (at least for the usage in this file).

Reply:
Mmmm I guess you're right! This could directly return the list. Thanks for the hints!

Reply:
I wonder about the performance impact of a generator vs a list. However, since for example `range` is since py3 also a generator, I guess there's no big difference.

Reply:
Is there any performance critical section in this code? It looks to me as the ranges are calculated once upfront and then you submit everything to spark. Is this correct?

Reply:
Yes indeed, everything is computed only once so it's not a critical section. You can still check the benchmark comparison I posted a few comments above :smiley: 
About generator vs list it shouldn't be an issue at all for the same reason. The real difference here is that I create the list at line 255 mainly to have the debug sentence the next line. Then I use that list in all the following list comprehensions, whereas I would need to call `_n_even_chunks` inside each of them hadn't I created the list beforehand.

As I was suggesting in a comment above, there's a possibility to shrink all the list comprehensions into a single one. That avoids some extra loops, even though I showed that the performance seems to not be affected for this usecase (I tried with a list of two files and few hundred clusters per each file). If in the end we go for the single list comprehension, then I could avoid creating the list and just call the generator in the list comprehension